homeassistant/components/motion_blinds/sensor.py

```diff
@@ -0,0 +1,178 @@
+"""Support for Motion Blinds sensors."""
+import logging
+
+from motionblinds import BlindType
+
+from homeassistant.const import (
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_SIGNAL_STRENGTH,
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, KEY_COORDINATOR, KEY_GATEWAY
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_BATTERY_VOLTAGE = "battery_voltage"
+TYPE_BLIND = "blind"
+TYPE_GATEWAY = "gateway"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Perform the setup for Motion Blinds."""
+    entities = []
+    motion_gateway = hass.data[DOMAIN][config_entry.entry_id][KEY_GATEWAY]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+
+    for blind in motion_gateway.device_list.values():
+        entities.append(MotionSignalStrengthSensor(coordinator, blind, TYPE_BLIND))
+        if blind.type == BlindType.TopDownBottomUp:
+            entities.append(MotionTDBUBatterySensor(coordinator, blind, "Bottom"))
+            entities.append(MotionTDBUBatterySensor(coordinator, blind, "Top"))
+        elif blind.battery_voltage > 0:
+            # Only add battery powered blinds
+            entities.append(MotionBatterySensor(coordinator, blind))
+
+    entities.append(
+        MotionSignalStrengthSensor(coordinator, motion_gateway, TYPE_GATEWAY)
+    )
+
+    async_add_entities(entities)
+
+
+class MotionBatterySensor(CoordinatorEntity, Entity):
+    """
+    Representation of a Motion Battery Sensor.
+
+    Updates are done by the cover platform.
+    """
+
+    def __init__(self, coordinator, blind):
+        """Initialize the Motion Battery Sensor."""
+        super().__init__(coordinator)
+
+        self._blind = blind
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the blind."""
+        return f"{self._blind.mac}-battery"
+
+    @property
+    def device_info(self):
+        """Return the device info of the blind."""
+        return {"identifiers": {(DOMAIN, self._blind.mac)}}
+
+    @property
+    def name(self):
+        """Return the name of the blind battery sensor."""
+        return f"{self._blind.blind_type}-battery-{self._blind.mac}"
```

Comment:
Is there a more human readable name that can be used?

Reply:
see my comment above